src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentDesignTimeNodeWriter.cs

```diff
@@ -1230,7 +1233,7 @@ protected override void WriteReferenceCaptureInnards(CodeRenderingContext contex
     public override void WriteRenderMode(CodeRenderingContext context, RenderModeIntermediateNode node)
     {
         // Looks like:
-        // global::Microsoft.AspNetCore.Components.IComponentRenderMode __renderMode = expression;
+        // __o = (global::Microsoft.AspNetCore.Components.IComponentRenderMode)(expression);
```

Comment:
What code causes `_o` to be declared?

Reply:
`__o` is introduced by the [Design time directive pass](https://source.razor.fyi/#Microsoft.AspNetCore.Razor.Language/Extensions/DesignTimeDirectivePass.cs,55)

When in design time mode, we can guarantee that its already declared.